### PR TITLE
Prometheus: Add multi-word search and higlight to metric selector

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

This PR adds metric search with highlights and multi-word search on words split with space (so searching `template failures` would match a metric name `prometheus_template_text_expansion_failures_total`. 

- The highlighter and styling is the same as we use to highlight string matches in log lines
- I was thinking about trying to implement fuzzy search, but it seems like a bigger issue that we can (possibly) tackle in the future, if users will want it and multi-word won't be enough. For fuzzy search, we could use `fuzzyMatch` algorithm from `grafana/ui`, but the issue is with ordering as fuzzyMatch produces distance that can be used for sorting, but `react-select` that we use doesn't support custom sorting, which means that we would have to do it outside of Select component. 

https://user-images.githubusercontent.com/30407135/152765010-3f259622-2821-4ab7-a6a2-55f003b8d022.mov

**Which issue(s) this PR fixes**:

Part of https://github.com/grafana/grafana/issues/44253

**To test**:

1. Enable `promQueryBuilder` feature toggle in custom.ini
2. In metric dropdown search for metrics. The search should find words in metrics. Use space a to specify each word.

### CODE REVIEW

**Comment by torkelo** [comment]
Nice!!

**Comment by torkelo** [comment]
searching on `chunk_tot_bytes` I am not sure should match `cortex_chunk_store_fetched_chunk_bytes_total`   (as there no white space in the search string)

maybe it would be ok if there are no hits for `chunk_tot_bytes`,  think underscore is such a common part of metric names so we should split it on `_`

**Comment by torkelo** [comment]
searching on `chunk_tot_bytes` I am not sure should match `cortex_chunk_store_fetched_chunk_bytes_total`   (as there no white space in the search string)

maybe it would be ok if there are no hits for `chunk_tot_bytes`,  think underscore is such a common part of metric names so we should split it on `_`

**Comment by ivanahuckova** [comment]
>searching on chunk_tot_bytes I am not sure should match cortex_chunk_store_fetched_chunk_bytes_total (as there no white space in the search string)

Yeah, I kind of agree. We can always extend searching regex later if we want. My logic was that `_`  is a common pattern where `_` splits the words. So it would be easier for me to find multi-word search feature when I would be writing down the metric I am searching for and realised I can match multiple words split with `_` . But maybe it is because I was developing this feature and it would be confusing for other users. I'll change it.

File: public/app/plugins/datasource/prometheus/querybuilder/components/MetricSelect.tsx [hunk 1]
```diff
@@ -12,12 +16,39 @@ export interface Props {
...
+      return (
+        <Highlighter
+          searchWords={meta.inputValue.split(splitSeparator)}
+          textToHighlight={option.label ?? ''}
+          highlightClassName={styles.hightlight}
+        />
+      );
+    },
+    [styles.hightlight]
+  );
```
Comment (torkelo) [inline public/app/plugins/datasource/prometheus/querybuilder/components/MetricSelect.tsx:50]: would be interesting topic for frontend weekly or #frontend and #query-builders,  feels like this feature (Multi word search and highlight) could great in other query editors .

**Review by torkelo — APPROVED** [review]
Awesome work on this! think this will be very appreciated feature

File: public/app/plugins/datasource/prometheus/querybuilder/components/MetricSelect.tsx (hunk 1 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/prometheus/querybuilder/components/MetricSelect.tsx:50]: I've added it to frontend weekly next week's agenda: https://docs.google.com/document/d/1LmS2esCEGI1eacQMgg-0ZyRjg84kRFw6Z1XIqkCeOIc/edit#heading=h.6bso5590olwu

**Pull request merged**